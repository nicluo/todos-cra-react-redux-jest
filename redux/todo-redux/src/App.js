import React from 'react';
import Header from './Header';
import TodoList from "./TodoList";
import './App.css';

const App = () => (
  <div className="App">
    <Header />
    <TodoList/>
  </div>
);

export default App;
