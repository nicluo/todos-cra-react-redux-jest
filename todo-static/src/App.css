.SubmitButton {
    background: none;
    border: none;
    outline: none;
    width: 30px;
    height: 30px;
    margin-right: 30px;
    border-radius: 50%;
    cursor: pointer;
}

.SubmitButton:hover {
    background: #ddd;
}

.Symbol {
    width: 30px;
}

.Checkbox {
    width: 30px;
    height: 30px;
    margin-right: 30px;
    background-color: white;
    border-radius: 50%;
    border: 1px solid #ddd;
    -webkit-appearance: none;
    outline: none;
    cursor: pointer;
}

.Checkbox:checked {
    background-image: url('Check.svg');
    background-size: 30px;
}

.Input {
    background: none;
    outline: none;
    border: none;
    border-bottom: 1px solid #ddd;
    padding: 4px;
    flex-grow: 1;
}
