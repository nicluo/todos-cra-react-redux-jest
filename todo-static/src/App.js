import React, { Component } from 'react';
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <header className="Header bg-black-90 w-100 ph3 pv3 pv4-ns ph4-m ph5-l tc">
          <nav className="f6 fw6 ttu tracked">
            <a className="link dim white dib" href="#" title="Todo">Todo</a>
          </nav>
        </header>

        <div className="TodoList ph3 mv4">
          <div className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3">
            <div className="flex items-center pa3 f4 f4-ns lh-copy">
              <input type="checkbox" className="Checkbox" checked />
              <div className="dib">Start With A Mock</div>
            </div>
          </div>
          <div className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3">
            <div className="flex items-center pa3 f4 f4-ns lh-copy">
              <input type="checkbox" className="Checkbox" />
              <div className="dib">Step 1: Break The UI Into A Component Hierarchy</div>
            </div>
          </div>
          <div className="TodoItem center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3">
            <div className="flex items-center pa3 f4 f4-ns lh-copy">
              <input type="checkbox" className="Checkbox" />
              <div className="dib">Step 2: Build A Static Version in React</div>
            </div>
          </div>
          <div className="TodoForm center mw-100 mw8-ns ba bw1 b--black-40 br2 mv3">
            <form>
              <div className="flex items-center pa3 f4 f4-ns lh-copy">
                <button type="submit" className="SubmitButton f3 dt">
                  <div className="Symbol dtc">+</div>
                </button>
                <input type="text" className="Input mw-100"/>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

export default App;
