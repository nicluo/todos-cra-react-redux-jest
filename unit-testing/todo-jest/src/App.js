import React, { Component } from 'react';
import Header from './Header';
import TodoList from "./TodoList";
import './App.css';

class App extends Component {
  render() {
    return (
      <div className="App">
        <Header />
        <TodoList/>
      </div>
    );
  }
}

export default App;
