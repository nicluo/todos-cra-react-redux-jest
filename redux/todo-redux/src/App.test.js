import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import App from './App';

it('Renders without crashing', () => {
  const component = shallow(
    <App />
  );
  expect(shallowToJson(component)).toMatchSnapshot();
});