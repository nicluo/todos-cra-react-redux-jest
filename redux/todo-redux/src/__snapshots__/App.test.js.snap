// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders without crashing 1`] = `
<div
  className="App"
>
  <Header />
  <TodoList />
</div>
`;
